files.go

```diff
@@ -0,0 +1,159 @@
+package dotfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+const (
+	dotfileDir = ".dotfile"
+	dotfile    = "files.json"
+)
+
+type trackedFile struct {
+	Path    string   `json:"path"`
+	Commits []string `json:"commits"`
+}
+
+// Init sets up a file for dotfile to track.
+func Init(path string, fileName string) error {
```

Comment:
This is a pretty long function; could you split it out into a few well-defined helper functions?